src/database.ts

```diff
@@ -55,24 +66,109 @@ import {
   TransactionRunner,
 } from './transaction-runner';
 
-export interface GetDatabaseOptions {
-  autoCreate?: boolean;
+import {google} from '../proto/spanner';
+import {Schema, RequestCallback, PagedRequest} from './common';
+import {ServiceError, CallOptions} from 'grpc';
+import {Readable, Transform, Duplex} from 'stream';
+import {PreciseDate} from '@google-cloud/precise-date';
+
+interface CreateBatchTransactionCallback {
+  (
+    err?: ServiceError | null,
+    transaction?: BatchTransaction | null,
+    apiResponse?: google.spanner.v1.ITransaction | google.spanner.v1.ISession
+  ): void;
 }
-export type DatabaseResponse = [Database, r.Response];
-export interface DatabaseCallback {
-  (err: Error | null, database?: Database, apiResponse?: r.Response): void;
+
+type CreateBatchTransactionResponse = [
+  BatchTransaction,
+  google.spanner.v1.ITransaction | google.spanner.v1.ISession
+];
+interface Cancellable {
+  cancel(): void;
 }
+type CancellableDuplex = Duplex & Cancellable;
```

Comment:
I think you can pull the stream types from `grpc`.

Reply:
used `ClientDuplexStream`